internal/elrond/elrond_soak.go

```diff
@@ -0,0 +1,63 @@
+package elrond
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/mattermost/elrond/model"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	pmodel "github.com/prometheus/common/model"
+	"github.com/sirupsen/logrus"
+)
+
+func checkSLOs(ring *model.Ring, thanosURL string, logger *logrus.Entry) error {
+	results, err := querySLOMetrics(ring, thanosURL, time.Now(), logger)
+	for _, result := range results {
+		if len(result) > 0 {
+			for _, rawMetric := range result {
+				sloService, ok := rawMetric.Metric["slo_service"]
+				if !ok {
+					continue
+				}
+				return errors.Errorf("SLO service %s is showing a high API error rate. Soaking failed, stopping release", sloService)
+			}
+		} else {
+			logger.Info("SLO metric checks looking good...")
+		}
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "failed to query thanos")
+	}
```

Comment:
this should be higher otherwise the for loop will break, right?

Reply:
> this should be higher otherwise the for loop will break, right?

Nice catch thanks dude